logict.cabal

```diff
@@ -35,7 +35,7 @@ library
   ghc-options: -O2 -Wall
   build-depends:
     base >=2 && <5,
-    mtl >=2 && <2.3
+    mtl >=2.2.1 && <2.3
```

Comment:
This does not look right. I'd rather import `Control.Monad` explicitly, or replace `pure` by `return` and `<$>` by `fmap`, if that's the problem.

Reply:
The problem here is that Control.Monad.Except wasn't introduced until 2.2.1.  There seems to be some odd versioning where 7.4 can use 2.2.1, and 7.8 can (although it doesn't default to it) but 7.6 does not and trying to use 2.2.1 with 7.6 conflicts with the transformers version in 7.6.  My thinking is to use more CPP to exclude the import of Control.Monad.Except and the associated test for 7.6 and 7.8 (and then remove this lower-bound constraint on mtl) if that sounds OK to you.